### PR TITLE
Userland: Pass Gfx::IntPoint by value

### PR DESCRIPTION
**Pull request by frhun**
This covers the cases where this could be done without much effort, only removing the `&` from some `auto` declarations.
There are other cases left where it is a bit more involved than just using find and replace.

### CODE REVIEW

**Comment by frhun** [comment]
This is in essence the same as https://github.com/SerenityOS/serenity/commit/8b1108e4858f797c9216dc8ae4a3918ad50c73b4 , done for the same reasons.

**Comment by tomuta** [comment]
What's the rationale here? It seems generally cheaper to pass a pointer to an object rather than running the copy constructor to duplicate an object on the stack? It feels like this would be the case even for very small objects like this. Or what am I missing?

**Comment by frhun** [comment]
> What's the rationale here? It seems generally cheaper to pass a pointer to an object rather than running the copy constructor to duplicate an object on the stack? It feels like this would be the case even for very small objects like this. Or what am I missing?

It's not "running the copy constructor", rather just copying/moving the two ints from one register to another. And maybe not even that.
In fact, one 64 bit pointer and two 32 bit ints are the same size, only that one needs referencing and dereferencing, while the other doesn't.

**Comment by MacDue** [comment]
(The rational here is pretty much the same as in: https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/)

**Comment by GMTA** [comment]
Do you have performance measurements to show that passing them by value makes a significant difference?

This seems more like a code style issue, but there is nothing in this PR to prevent them being passed as a reference in the future.

**Comment by tomuta** [comment]
I don't find this overly convincing tbh. In theory it may be (almost) equivalent for very simple types, but I still believe that it's more efficient to pass a const reference. I suspect it falls apart if the class does have a copy constructor that is implemented in a cpp file rather than directly in the header (not sure how good LTO is in this regard).

**Comment by frhun** [comment]
> I don't find this overly convincing tbh. In theory it may be (almost) equivalent for very simple types, but I still believe that it's more efficient to pass a const reference. I suspect it falls apart if the class does have a copy constructor that is implemented in a cpp file rather than directly in the header (not sure how good LTO is in this regard).

But IntPoint doesn't.

**Comment by ldm5180** [comment]
Avoiding the pointer indirection is likely a performance win. Can you provide some measurements?

**Comment by ldm5180** [comment]
I don't think this is a stylistic change. This seems to be best practices to me. Passing small objects by value is generally considered a good thing. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in

File: Userland/Applications/Browser/Tab.cpp [hunk 1]
```diff
@@ -257,7 +257,7 @@ Tab::Tab(BrowserWindow& window)
     m_link_context_menu->add_separator();
     m_link_context_menu->add_action(window.inspect_dom_node_action());
 
-    view().on_link_context_menu_request = [this](auto& url, auto& screen_position) {
+    view().on_link_context_menu_request = [this](auto& url, auto screen_position) {
```
Comment (Hendiadyoin1) [inline Userland/Applications/Browser/Tab.cpp:260]: Not something you have to do, just a general thought:
Maybe spelling the types here would help, or make it more clear what is exactly happening with these lambdas

Comment (frhun) [reply Userland/Applications/Browser/Tab.cpp:260]: AFAIK it is the recommended coding style in SerenityOS to use auto where possible.

**Comment by frhun** [comment]
> Avoiding the pointer indirection is likely a performance win. Can you provide some measurements?

Sadly no, it doesn't help that it's basically all in UI code. Getting good reproducible data out of this code would be certainly possible but I would have to write a drawing performance benchmark.

**Comment by Hendiadyoin1** [comment]
> I would have to write a drawing performance benchmark.

We have a libGFX benchmark iirc

**Pull request closed without merging**